sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -475,6 +496,10 @@ public DataSource apply(Void input) {
 
       abstract Builder<T> setRowMapper(RowMapper<T> rowMapper);
 
+      abstract Builder<T> setBeamSchemaRowMapper(BeamSchemaRowMapper<T> beamSchemaRowMapper);
+
+      abstract Builder<T> setWithBeamSchema(Boolean autoGenerateBeamSchema);
```

Comment:
@charithe We  found that the AutoValue class was marking the primitive types as mandatory if we use the primitive type. The intention to use the boxed type was to avoid this mandatory validation because then we had to explicitly provide "false" just like it does with `.setOutputParallelization(true)`.

We can change this to primitive type but then we have to pass `setWithBeamSchema(false)` explicitly. What do you say?

Reply:
@charithe, The convention we followed is same. Builder methods are implemented with `with` which internally calls the `setWith` method

Reply:
@JawadHyder @salmanVD  Are we getting rid of setWithBeamSchema, hence the discussion about primitive vs boxed booleans is moot?

Reply:
@shehzaadn-vd With latest push, this property is removed